tests/integration/targets/ec2_carrier_gateway/tasks/main.yml

```diff
@@ -0,0 +1,162 @@
+---
+- name: 'ec2_carrier_gateway integration tests'
+  collections:
+    - community.aws
+  module_defaults:
+    group/aws:
+      aws_access_key: '{{ aws_access_key }}'
+      aws_secret_key: '{{ aws_secret_key }}'
+      security_token: '{{ security_token | default(omit) }}'
+      region: '{{ aws_region }}'
+  block:
+
+    # ============================================================
+    - debug: msg="Setting up test dependencies"
+
+    - name: create a VPC
+      ec2_vpc_net:
+        name: "{{ vpc_name }}-{{ item }}"
+        state: present
+        cidr_block: "{{ vpc_cidr }}"
+        tags:
+          Description: "Created by ansible-test for CAGW tests"
+      register: vpc_result
+      loop: [1]
+
+    - name: use set fact for vpc ids
+      set_fact:
+        vpc_id_1: '{{ vpc_result.results.0.vpc.id }}'
+
+    # ============================================================
+    - debug: msg="Running tests"
+
+    - name: create carrier gateway and attach it to vpc
```

Comment:
@mtulio can you also use ` check_mode: True` in the tests for `ec2_carrier_gateway`?

Reply:
Done